packages/browser-agent-core/features/jserrors/instrument/index.js

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright 2020 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { handle } from '../../../common/event-emitter/handle'
+import { getRuntime } from '../../../common/config/config'
+import { now } from '../../../common/timing/now'
+import { getOrSet } from '../../../common/util/get-or-set'
+import { wrapRaf, wrapTimer, wrapEvents, wrapXhr } from '../../../common/wrap'
+import slice from 'lodash._slice'
+import './debug'
+import { FeatureBase } from '../../../common/util/feature-base'
+
+var NR_ERR_PROP = 'nr@seenError'
+
+export class Instrument extends FeatureBase {
+  constructor(agentIdentifier) {
+    super(agentIdentifier)
+    // skipNext counter to keep track of uncaught
+    // errors that will be the same as caught errors.
+    this.skipNext = 0
+    this.handleErrors = false
+    this.origOnerror = window.onerror
+    
+    const state = this
+
+    const agentRuntime = getRuntime(this.agentIdentifier);
+    agentRuntime.features.err = true;   // declare that we are using err instrumentation
+
+
+    // // NREUM.debug("initialize js-errors instrument!", state.agentIdentifier)
+
+    state.ee.on('fn-start', function (args, obj, methodName) {
+      if (state.handleErrors) state.skipNext += 1
+    })
+
+    state.ee.on('fn-err', function (args, obj, err) {
+      if (state.handleErrors && !err[NR_ERR_PROP]) {
+        getOrSet(err, NR_ERR_PROP, function getVal() {
+          return true
+        })
+        this.thrown = true
+        notice(err, undefined, state.ee)
+      }
+    })
+
+    state.ee.on('fn-end', function () {
+      if (!state.handleErrors) return
+      if (!this.thrown && state.skipNext > 0) state.skipNext -= 1
+    })
+
+    state.ee.on('internal-error', (e) => {
+      handle('ierr', [e, now(), true], undefined, undefined, state.ee)
+    })
+
+    // Declare that we are using err instrumentation
+    // require('./debug')
+
+    const prevOnError = window.onerror
+    window.onerror = (...args) => {
+      if (prevOnError) prevOnError(...args)
+      this.onerrorHandler(...args)
+      return false
+    }
+
+    try {
+      window.addEventListener('unhandledrejection', (e) => {
```

Comment:
What is this try/catch supposed to do and where did it come from? It looks like it was added in the Dev commit but I don't see it in the original code.

Reply:
I **think** this was added in the original NPM work, to catch unhandled promise rejections.  This is likely unnecessary now that all the `wrap` files have been implemented, but it is worth checking.